From 649bf7c233fd55859f5d0b294b5130531c520505 Mon Sep 17 00:00:00 2001
From: Tim Murray <timmurray@google.com>
Date: Wed, 21 Mar 2018 15:42:09 -0700
Subject: [PATCH 6/8] fpc: keep fpc in system-background

The fingerprint HAL is insensitive to increased CPU throughput, but it also
has a tendency to spin while waiting for FP hardware. Limit FPC to the
system-background cpuset in order to avoid increased power consumption
when accidentally touching the fingerprint sensor.

bug 76115243
Test: fingerprint in system-background cpuset on taimen

Change-Id: Iaffe6f63bd76b7a1c4acaf0cae980840af515961
---
 .../2.0/android.hardware.biometrics.fingerprint@2.0-service.rc   | 1 +
 1 file changed, 1 insertion(+)

diff --git a/biometrics/fingerprint/2.0/android.hardware.biometrics.fingerprint@2.0-service.rc b/biometrics/fingerprint/2.0/android.hardware.biometrics.fingerprint@2.0-service.rc
index 076d904..26c7170 100644
--- a/biometrics/fingerprint/2.0/android.hardware.biometrics.fingerprint@2.0-service.rc
+++ b/biometrics/fingerprint/2.0/android.hardware.biometrics.fingerprint@2.0-service.rc
@@ -5,3 +5,4 @@ service fps_hal /vendor/bin/hw/android.hardware.biometrics.fingerprint@2.0-servi
     class late_start
     user system
     group system input
+    writepid /dev/cpuset/system-background/tasks
-- 
2.17.1

